Makefile.kokkos

```diff
@@ -1212,6 +1214,11 @@ ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX942), 1)
   tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942\"")
   KOKKOS_INTERNAL_AMD_ARCH_FLAG := --offload-arch=gfx942
 endif
+ifeq ($(KOKKOS_INTERNAL_USE_ARCH_AMD_GFX942_APU), 1)
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GFX942_APU")
+  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942\"")
```

Comment:
```suggestion
  tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_AMD_GPU \"gfx942_APU\"")
```
or is it referring to the base architecture?

Reply:
I checked the cmake file and I think it's using the flag there. So I kept it consistent.

Reply:
https://github.com/kokkos/kokkos/blob/54904946f91db5abaa8afd020c47723a4483fb58/cmake/kokkos_arch.cmake#L1287C5-L1287C39